usaspending_api/bulk_download/v2/views.py

```diff
@@ -0,0 +1,311 @@
+                .values('federal_account_name', 'federal_account_id')
+        return Response(response_data)
+
+
+class BulkDownloadAwardsViewSet(BaseDownloadViewSet):
+    """Generate bulk download for awards"""
+
+    def process_filters(self, filters, award_level):
+        """Filter function for Bulk Download Award Generation"""
+
+        for required_param in ['award_types', 'agency', 'sub_agency', 'date_type', 'date_range']:
+            if required_param not in filters:
+                raise InvalidParameterException('{} filter not provided'.format(required_param))
+
+        and_queryset_filters = {}
+
+        # Adding award type filter
+        award_types = []
+        if not isinstance(filters['award_types'], list):
+            raise InvalidParameterException('award_types parameter not provided as a list')
+        for award_type in filters['award_types']:
+            if award_type in award_type_mappings:
+                award_types.extend(award_type_mappings[award_type])
+            else:
+                raise InvalidParameterException('Invalid parameter for award_types: {}'.format(award_type))
+        # if the filter is calling everything, just remove the filter, save on the query performance
+        if set(award_types) != set(itertools.chain(*award_type_mappings.values())):
+            and_queryset_filters['{}__in'.format(value_mappings[award_level]['type'])] = award_types
+
+        # Adding date range filters
+        # Get the date type attribute
+        date_attribute = None
+        if filters['date_type'] == 'action_date':
+            date_attribute = value_mappings[award_level]['action_date']
+        elif filters['date_type'] == 'last_modified_date':
+            date_attribute = value_mappings[award_level]['last_modified_date']
+        else:
+            raise InvalidParameterException('Invalid parameter for date_type: {}'.format(filters['date_type']))
+        # Get the date ranges
+        if not isinstance(filters['date_range'], dict):
+            raise InvalidParameterException('date_range parameter not provided as an object')
+        if 'start_date' in filters['date_range']:
+            and_queryset_filters['{}__gte'.format(date_attribute)] = filters['date_range']['start_date']
+        if 'end_date' in filters['date_range']:
+            and_queryset_filters['{}__lte'.format(date_attribute)] = filters['date_range']['end_date']
+
+        # Agencies are to be OR'd together and then AND'd to the major query
+        or_queryset = (Q(awarding_agency_id=filters['agency']) |
+                                   Q(funding_agency_id=filters['agency']))
+        if 'sub_agency' in filters:
+            or_queryset = (or_queryset & (Q(awarding_agency_id=filters['sub_agency']) |
+                                          Q(funding_agency_id=filters['sub_agency'])))
+
+        # Put it all together
+        table = value_mappings[award_level]['table']
+        if and_queryset_filters and or_queryset:
```

Comment:
Doesn't this evaluated or_queryset and hit the DB?

Reply:
Great catch.